core/include/jiminy/core/robot/Robot.h

```diff
@@ -138,6 +171,8 @@ namespace jiminy
 
         std::vector<std::string> motorTorqueFieldnames_;    ///< Fieldnames of the torques of the motors
 
+        std::unordered_map<std::string, std::shared_ptr<AbstractConstraint>> constraintsHolder_;
```

Comment:
static_map instead ? It is more efficient. Use only unordered_map if you need to find elements by key repeatedly.

Reply:
You mean mpa ? static_map seems to be part of cusom libraries but not in std.

Reply:
yes, our library :p you can check type.h :)